### PR TITLE
Olivekl docs cleanup

### PR DESCRIPTION
**Pull request by olivekl**
General cleanup: remove TODOs, fix broken link, move a link earlier, fix typos, remove or reword a few sentences

### CODE REVIEW

File: docs/developing_clusterfuzzlite.md [hunk 1]
```diff
@@ -13,13 +13,6 @@ permalink: /developing-clusterfuzzlite/
 {:toc}
 ---
 
-TODO
```
Comment (jonathanmetzman) [inline docs/developing_clusterfuzzlite.md:16]: Can you leave these for now? I'm going to fix them today.

File: docs/overview.md [hunk 2]
```diff
@@ -22,9 +22,7 @@ The program that creates the inputs is called a fuzzer.
 Fuzzing is highly effective at finding bugs missed by manually written tests,
 code review, or auditing.
 Fuzzing has found thousands of bugs in mature software such as Chrome, OpenSSL,
-and Curl.
-If fuzzing isn't finding bugs in your code, your code is probably not bug free,
-you are probably not fuzzing well.
+and Curl. When done well, fuzzing is able to find bugs in virtually any code.
```
Comment (jonathanmetzman) [inline docs/overview.md:25]: One thing I try to do when writing markdown is to start every sentence on a new line.
This makes merge conflicts less likely.
It's not necessary but I think it's a good idea and can make things a little more readable IMO.

**Review by jonathanmetzman — APPROVED** [review]
LGTM except for that one comment about my TODOs

Thanks!

File: docs/developing_clusterfuzzlite.md (hunk 1 above)
Comment (olivekl) [reply docs/developing_clusterfuzzlite.md]: Reinstated.

File: docs/overview.md (hunk 2 above)
Comment (olivekl) [reply docs/overview.md]: Sure, thanks for the tip! Will do going forward.

**Review by jonathanmetzman — APPROVED** [review]
LGTM

File: docs/overview.md (hunk 2 above)
Comment (jonathanmetzman) [reply docs/overview.md]: Thanks!

**Pull request merged**